apps/web/test/lib/getSchedule.test.ts

```diff
@@ -574,20 +576,20 @@ describe("getSchedule", () => {
         input: {
           eventTypeId: 1,
           eventTypeSlug: "",
-          startTime: `${plus2DateString}T18:30:00.000Z`,
-          endTime: `${plus3DateString}T18:29:59.999Z`,
+          startTime: `${plus2DateString}T18:30:00.000Z`, // Day 3 0 AM IST
+          endTime: `${plus3DateString}T18:29:59.999Z`, // Day 3 23:59 PM IST
           timeZone: Timezones["+5:30"],
           isTeamEvent: false,
         },
       });
 
       expect(scheduleForEventOnADayWithNonCalBooking).toHaveTimeSlots(
         [
-          // `04:00:00.000Z`, // - 4 AM is booked
-          // `06:00:00.000Z`, // - 6 AM is not available because 08:00AM slot has a `beforeEventBuffer`
-          `08:00:00.000Z`, // - 8 AM is available because of availability of 06:00 - 07:59
```

Comment:
Earlier this was passing (in UTC) because we were subtracting the busyTimes (UTC) from dateRanges (any tz).

https://github.com/calcom/cal.com/blob/0a795f5363b82fd5a5b6ffa1a4fa31a035a3f44f/packages/core/getUserAvailability.ts#L334-L346

So, we were getting somethings like this.

![Explanation](https://github.com/calcom/cal.com/assets/86384652/9a2a0d5d-a3bb-4dbc-bd0a-b20bed3d2bc2)

And this was being sent to `getSlots` fn. 

While generating the slots, the value of `minute` for slots 6 AM(UTC) & further were being read as `0` because the `rangeStart` was in UTC format. So, the value of `minute` was fitting perfectly with the slot interval.